test/manipulation.test.js

```diff
@@ -193,6 +297,29 @@ describe('manipulation', function () {
       });
     });
 
+    it('should create batch of objects with beforeCreate (promise variant)', function(done) {
+      Person.beforeCreate = function(next, data) {
+        if (data && data.name === 'A') {
+          return next(null, {id: 'a', name: 'A'});
+        } else {
+          return next();
+        }
+      };
+      var batch = [
+        {name: 'A'},
+        {name: 'B'},
+        undefined
+      ];
+      Person.create(batch)
+      .then(function(ps) {
+        should.exist(ps);
+        ps.should.be.instanceOf(Array);
+        ps.should.have.lengthOf(batch.length);
+        ps[0].should.be.eql({id: 'a', name: 'A'});
+        done();
+      }).catch(done);
+    });
```

Comment:
> any ideas on how to test promise support without duplicating so many tests?

I'd say write only one test per each method to test the happy path. Since the implementation is using callback under the hood, I think it's ok to assume that it would work in other cases that are already covered by callback-based tests.